lib/src/value/string.ts

```diff
@@ -126,7 +126,37 @@ export class SassString extends Value {
   }
 
   toString(): string {
-    return this.hasQuotes ? `"${this.text}"` : this.text;
+    if (!this.hasQuotes)
+      return this.text.replace(/\0/g, '\uFFFD').replace(/\n */g, ' ');
+
+    // https://drafts.csswg.org/cssom/#serialize-a-string
+    let buffer = '"';
+    for (const character of this.text) {
+      if (character === '\0') {
+        // If the character is NULL (U+0000), then the REPLACEMENT CHARACTER
+        // (U+FFFD).
+        buffer += '\uFFFD';
+      } else if (character === '\x22' || character === '\x5C') {
+        // If the character is '"' (U+0022) or "\" (U+005C), then the escaped
+        // character.
+        buffer += '\x5C' + character;
+      } else if (character < '\x20' || character === '\x7F') {
+        // If the character is in the range [\1-\1f] (U+0001 to U+001F) or is
+        // U+007F, then the character escaped as code point.
+        //
+        // To escape a character as code point means to create a string of "\"
+        // (U+005C), followed by the Unicode code point as the smallest possible
+        // number of hexadecimal digits in the range 0-9 a-f (U+0030 to U+0039
+        // and U+0061 to U+0066) to represent the code point in base 16,
+        // followed by a single SPACE (U+0020).
+        buffer += '\x5C' + character.codePointAt(0)!.toString(16) + '\x20';
+      } else {
+        // Otherwise, the character itself.
+        buffer += character;
+      }
+    }
+    buffer += '"';
+    return buffer;
```

Comment:
This is not exactly the same as how dart-sass would serialize quoted string, but it is a documented reference implementation. The end result is the same once the stylesheet is parsed by the browser.

The differences are:

1. dart-sass uses double quote or single quote depends on which kind of quote is present in the raw string. This implementation always use double quote as described by the linked csswg documentation.
2. dart-sass does not escape "TAB (U+0009)". This implementation escapes it as described by the linked csswg documentation.
3. dart-sass checks the next character and omits the trailing space if possible when escaping a character as code point. This implementation always produces s space as described by the linked csswg documentation.